tests/bfd/test_bfd_static_route.py

```diff
@@ -0,0 +1,1369 @@
+            action = "startup"
+            self.control_interface_state(src_dut, src_asic, bfd_base_instance, interface, action)
+        
+        # Verification of BFD session state.
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(dst_dut, dst_dut_nexthops.values(), dst_asic, "Up"))
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(src_dut, src_dut_nexthops.values(), src_asic, "Up"))
+
+        # Verify that corresponding static route has been added on both duts
+        logger.info("BFD & Static route verifications")
+        self.verify_static_route(request, dst_asic, dst_prefix, dst_dut, dst_dut_nexthops, "Route Addition", bfd_base_instance, version)
+        self.verify_static_route(request, src_asic, src_prefix, src_dut, src_dut_nexthops, "Route Addition", bfd_base_instance, version)
+    
+    def test_bfd_portchannel_member_flap_ipv4(self, duthost, request, duthosts, tbinfo, get_src_dst_asic_and_duts, bfd_base_instance, bfd_cleanup_db):
+        version = "ipv4"
+
+        request.config.interface_shutdown = True
+
+        # Selecting source, destination dut & prefix & BFD status verification for all nexthops
+        logger.info("Selecting Source dut, destination dut, source asic, destination asic, source prefix, destination prefix")
+        src_asic, dst_asic, src_dut, dst_dut, src_dut_nexthops, dst_dut_nexthops, src_prefix, dst_prefix = self.select_src_dst_dut_with_asic(request, get_src_dst_asic_and_duts, bfd_base_instance, version)
+        
+        # Creation of BFD
+        logger.info("BFD addition on source dut")   
+        bfd_base_instance.add_bfd(src_asic.asic_index, src_prefix, src_dut)
+        logger.info("BFD addition on destination dut")   
+        bfd_base_instance.add_bfd(dst_asic.asic_index, dst_prefix, dst_dut)
+
+        # Verification of BFD session state.
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(dst_dut, dst_dut_nexthops.values(), dst_asic, "Up"))
+        assert wait_until(180, 10, 0, lambda: bfd_base_instance.verify_bfd_state(src_dut, src_dut_nexthops.values(), src_asic, "Up"))
+
+        # Extract portchannel interfaces on src
+        list_of_portchannels_on_src = dst_dut_nexthops.keys()
+        request.config.portchannels_on_dut = "src"
+        request.config.selected_portchannels = list_of_portchannels_on_src
+        
+        #Shutdown PortChannel members
+        for portchannel_interface in list_of_portchannels_on_src:
+            action = "shutdown"
+            list_of_portchannel_members_on_src = bfd_base_instance.extract_backend_portchannels(src_dut)[portchannel_interface]['members']
+            request.config.selected_portchannel_members = list_of_portchannel_members_on_src
+            for each_member in list_of_portchannel_members_on_src:
+                self.control_interface_state(src_dut, src_asic, bfd_base_instance, each_member, action)
```

Comment:
Potential enhancement:
When 1 member goes down, BFD should go down (2 member port channel - Q200) as well as for 3 member port channel. Verify BFD behaviour.